OPAL/br/src/main/scala/org/opalj/br/analyses/cg/EntryPointFinder.scala

```diff
@@ -347,4 +347,46 @@ object AllEntryPointsFinder extends EntryPointFinder {
             project.allProjectClassFiles.flatMap(_.methodsWithBody)
         else project.allMethodsWithBody
     }
+}
+
+/**
+ * The AndroidEntryPointFinder considers specific methods of app components as entry points.
+ * It does not work for androidx
+ *
+ * @author Tom Nikisch
+ */
+object AndroidEntryPointsFinder extends EntryPointFinder {
+
+    val activityEPS: List[String] = List("onCreate", "onRestart", "onStart", "onResume",
+        "onStop", "onDestroy", "onActivityResult")
+    val serviceEPS: List[String] = List("onCreate", "onStartCommand", "onBind", "onStart")
+    val contentProviderEPS: List[String] = List("onCreate", "query", "insert", "update")
+    val locationListenerEPS: List[String] = List("onLocationChanged", "onProviderDisabled", "onProviderEnabled",
+        "onStatusChanged")
+    val onNmeaMessageListenerEPS: List[String] = List("onNmeaMessage")
+    val broadcastReceiverEPS: List[String] = List("onReceive")
+    val defaultEPS = Map("android/app/Activity" -> activityEPS, "android/app/Service" -> serviceEPS,
+        "android/content/ContentProvider" -> contentProviderEPS,
+        "android/location/LocationListener" -> locationListenerEPS,
+        "android/location/onNmeaMessageListener" -> onNmeaMessageListenerEPS,
+        "android/content/BroadcastReceiver" -> broadcastReceiverEPS)
+
+    override def collectEntryPoints(project: SomeProject): Traversable[Method] = {
+        val eps = ArrayBuffer.empty[Method]
+        for ((superClass, methodList) ← defaultEPS) {
+            eps ++= findEPS(ObjectType(superClass), methodList, project)
+        }
+        eps
+    }
+
+    def findEPS(ot: ObjectType, possibleEPS: List[String], project: SomeProject): ArrayBuffer[Method] = {
+        val eps = ArrayBuffer.empty[Method]
+        val classHierarchy = project.classHierarchy
+        classHierarchy.foreachSubclass(ot, project) { sc ⇒
+            for (pep ← possibleEPS; m ← sc.findMethod(pep) if m.body.isDefined && !eps.contains(m)) {
```

Comment:
While probably not too important here, it is quite imperformant to do contains on an ArrayBuffer. Since what you're obviously after is the semantics of a Set (i.e., you only add elements once anyway), why don't you use a Set or Set-like data structure in the first place?